diff --git a/CMake/Dependencies.cmake b/CMake/Dependencies.cmake
index c444861..4d311c2 100755
--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -285,21 +285,21 @@ if(ANDROID)
     set(Boost_COMPILER -gcc)
 endif()
 
-if (NOT Boost_FOUND)
-  set(Boost_ADDITIONAL_VERSIONS "1.57" "1.57.0" "1.56" "1.56.0" "1.55" "1.55.0" "1.54" "1.54.0" "1.53" "1.53.0" "1.52" "1.52.0" "1.51" "1.51.0" "1.50" "1.50.0" "1.49" "1.49.0" "1.48" "1.48.0" "1.47" "1.47.0" "1.46" "1.46.0" "1.45" "1.45.0" "1.44" "1.44.0" "1.42" "1.42.0" "1.41.0" "1.41" "1.40.0" "1.40")
-  # Components that need linking (NB does not include header-only components like bind)
-  set(OGRE_BOOST_COMPONENTS thread date_time)
-  find_package(Boost COMPONENTS ${OGRE_BOOST_COMPONENTS} QUIET)
-endif()
-if (NOT Boost_FOUND)
-	# Try again with the other type of libs
-	if(Boost_USE_STATIC_LIBS)
-		set(Boost_USE_STATIC_LIBS OFF)
-	else()
-		set(Boost_USE_STATIC_LIBS ON)
-	endif()
-	find_package(Boost COMPONENTS ${OGRE_BOOST_COMPONENTS} QUIET)
-endif()
+# if (NOT Boost_FOUND)
+#   set(Boost_ADDITIONAL_VERSIONS "1.57" "1.57.0" "1.56" "1.56.0" "1.55" "1.55.0" "1.54" "1.54.0" "1.53" "1.53.0" "1.52" "1.52.0" "1.51" "1.51.0" "1.50" "1.50.0" "1.49" "1.49.0" "1.48" "1.48.0" "1.47" "1.47.0" "1.46" "1.46.0" "1.45" "1.45.0" "1.44" "1.44.0" "1.42" "1.42.0" "1.41.0" "1.41" "1.40.0" "1.40")
+#   # Components that need linking (NB does not include header-only components like bind)
+#   set(OGRE_BOOST_COMPONENTS thread date_time)
+#   find_package(Boost COMPONENTS ${OGRE_BOOST_COMPONENTS} QUIET)
+# endif()
+# if (NOT Boost_FOUND)
+# 	# Try again with the other type of libs
+# 	if(Boost_USE_STATIC_LIBS)
+# 		set(Boost_USE_STATIC_LIBS OFF)
+# 	else()
+# 		set(Boost_USE_STATIC_LIBS ON)
+# 	endif()
+# 	find_package(Boost COMPONENTS ${OGRE_BOOST_COMPONENTS} QUIET)
+# endif()
 
 if(Boost_FOUND AND Boost_VERSION GREATER 104900)
     if(Boost_VERSION GREATER 105300)
