diff --git a/Components/Overlay/include/OgreFont.h b/Components/Overlay/include/OgreFont.h
index 44db2a6bb..286f7d083 100644
--- a/Components/Overlay/include/OgreFont.h
+++ b/Components/Overlay/include/OgreFont.h
@@ -27,6 +27,11 @@ THE SOFTWARE
 #ifndef _Font_H__
 #define _Font_H__
 
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 #include "OgreOverlayPrerequisites.h"
 #include "OgreResource.h"
 #include "OgreCommon.h"
@@ -420,4 +425,8 @@ namespace Ogre
     /** @} */
 }
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/Components/Overlay/include/OgreOverlayProfileSessionListener.h b/Components/Overlay/include/OgreOverlayProfileSessionListener.h
index 5b9cababd..dda7a57b3 100644
--- a/Components/Overlay/include/OgreOverlayProfileSessionListener.h
+++ b/Components/Overlay/include/OgreOverlayProfileSessionListener.h
@@ -28,6 +28,10 @@ THE SOFTWARE.
 
 #ifndef __OverlayProfileSessionListener_H__
 #define __OverlayProfileSessionListener_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
 
 #include "OgreOverlayPrerequisites.h"
 #include "OgreProfiler.h"
@@ -151,4 +155,8 @@ namespace Ogre  {
         DisplayMode mDisplayMode;
     };
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreAnimable.h b/OgreMain/include/OgreAnimable.h
index 755e76d00..38c86cfea 100644
--- a/OgreMain/include/OgreAnimable.h
+++ b/OgreMain/include/OgreAnimable.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __ANIMABLE_H__
 #define __ANIMABLE_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -293,5 +298,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreAutoParamDataSource.h b/OgreMain/include/OgreAutoParamDataSource.h
index bb35d6226..da6f2f18c 100644
--- a/OgreMain/include/OgreAutoParamDataSource.h
+++ b/OgreMain/include/OgreAutoParamDataSource.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __AutoParamDataSource_H_
 #define __AutoParamDataSource_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -269,5 +274,9 @@ namespace Ogre {
     /** @} */
     /** @} */
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreAxisAlignedBox.h b/OgreMain/include/OgreAxisAlignedBox.h
index c543a44d7..875333451 100644
--- a/OgreMain/include/OgreAxisAlignedBox.h
+++ b/OgreMain/include/OgreAxisAlignedBox.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __AxisAlignedBox_H_
 #define __AxisAlignedBox_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include <array>
 
@@ -808,5 +813,9 @@ namespace Ogre {
     /** @} */
     /** @} */
 } // namespace Ogre
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreBillboard.h b/OgreMain/include/OgreBillboard.h
index cd4420cf6..324b87b99 100644
--- a/OgreMain/include/OgreBillboard.h
+++ b/OgreMain/include/OgreBillboard.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __Billboard_H__
 #define __Billboard_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -217,5 +222,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreBillboardChain.h b/OgreMain/include/OgreBillboardChain.h
index 97cadc3f6..6105d88ea 100644
--- a/OgreMain/include/OgreBillboardChain.h
+++ b/OgreMain/include/OgreBillboardChain.h
@@ -31,6 +31,11 @@ THE SOFTWARE.
 
 #ifndef _BillboardChain_H__
 #define _BillboardChain_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -382,6 +387,8 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
 #endif
 
-
+#endif
diff --git a/OgreMain/include/OgreBillboardParticleRenderer.h b/OgreMain/include/OgreBillboardParticleRenderer.h
index 9187f687f..ce9606764 100644
--- a/OgreMain/include/OgreBillboardParticleRenderer.h
+++ b/OgreMain/include/OgreBillboardParticleRenderer.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __BillboardParticleRenderer_H__
 #define __BillboardParticleRenderer_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreParticleSystemRenderer.h"
@@ -221,5 +226,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif // __BillboardParticleRenderer_H__
 
diff --git a/OgreMain/include/OgreBillboardSet.h b/OgreMain/include/OgreBillboardSet.h
index c8ab575a1..cb3ef2e28 100644
--- a/OgreMain/include/OgreBillboardSet.h
+++ b/OgreMain/include/OgreBillboardSet.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __BillboardSet_H__
 #define __BillboardSet_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -824,5 +829,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __BillboardSet_H__
diff --git a/OgreMain/include/OgreBone.h b/OgreMain/include/OgreBone.h
index 8ffc52cf1..191e28ecc 100644
--- a/OgreMain/include/OgreBone.h
+++ b/OgreMain/include/OgreBone.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __Bone_H__
 #define __Bone_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreNode.h"
@@ -152,5 +157,9 @@ namespace Ogre
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreCamera.h b/OgreMain/include/OgreCamera.h
index 263c364c1..a1988cba8 100644
--- a/OgreMain/include/OgreCamera.h
+++ b/OgreMain/include/OgreCamera.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Camera_H__
 #define __Camera_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Default options
 #include "OgrePrerequisites.h"
@@ -654,5 +659,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __Camera_H__
diff --git a/OgreMain/include/OgreCompositionPass.h b/OgreMain/include/OgreCompositionPass.h
index 1c2196796..a9867d32c 100644
--- a/OgreMain/include/OgreCompositionPass.h
+++ b/OgreMain/include/OgreCompositionPass.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __CompositionPass_H__
 #define __CompositionPass_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreHeaderPrefix.h"
@@ -374,5 +379,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreConvexBody.h b/OgreMain/include/OgreConvexBody.h
index 4982280b6..57defd5d5 100644
--- a/OgreMain/include/OgreConvexBody.h
+++ b/OgreMain/include/OgreConvexBody.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 */
 #ifndef __ConvexBody_H__
 #define __ConvexBody_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgrePolygon.h"
@@ -257,5 +262,9 @@ namespace Ogre
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif 
 
diff --git a/OgreMain/include/OgreDualQuaternion.h b/OgreMain/include/OgreDualQuaternion.h
index e63c46cb8..cab7cf1df 100644
--- a/OgreMain/include/OgreDualQuaternion.h
+++ b/OgreMain/include/OgreDualQuaternion.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __DualQuaternion_H__
 #define __DualQuaternion_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMath.h"
@@ -180,5 +185,9 @@ namespace Ogre {
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif 
diff --git a/OgreMain/include/OgreEdgeListBuilder.h b/OgreMain/include/OgreEdgeListBuilder.h
index 2c86e620f..6f8961d6b 100644
--- a/OgreMain/include/OgreEdgeListBuilder.h
+++ b/OgreMain/include/OgreEdgeListBuilder.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __EdgeListBuilder_H__
 #define __EdgeListBuilder_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreRenderOperation.h"
@@ -274,5 +279,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreEntity.h b/OgreMain/include/OgreEntity.h
index d42d8a6b7..b6d405df0 100644
--- a/OgreMain/include/OgreEntity.h
+++ b/OgreMain/include/OgreEntity.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Entity_H__
 #define __Entity_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -889,5 +894,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __Entity_H__
diff --git a/OgreMain/include/OgreFrustum.h b/OgreMain/include/OgreFrustum.h
index 885d7c9f1..cc2cab2dc 100644
--- a/OgreMain/include/OgreFrustum.h
+++ b/OgreMain/include/OgreFrustum.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Frustum_H__
 #define __Frustum_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMovableObject.h"
@@ -679,5 +684,9 @@ namespace Ogre
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __Frustum_H__
diff --git a/OgreMain/include/OgreGpuProgram.h b/OgreMain/include/OgreGpuProgram.h
index 283cdf514..b10e86364 100644
--- a/OgreMain/include/OgreGpuProgram.h
+++ b/OgreMain/include/OgreGpuProgram.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __GpuProgram_H_
 #define __GpuProgram_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -444,5 +449,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreGpuProgramParams.h b/OgreMain/include/OgreGpuProgramParams.h
index 946b50d4f..f894e513a 100644
--- a/OgreMain/include/OgreGpuProgramParams.h
+++ b/OgreMain/include/OgreGpuProgramParams.h
@@ -27,6 +27,11 @@ Copyright (c) 2000-2014 Torus Knot Software Ltd
 */
 #ifndef __GpuProgramParams_H_
 #define __GpuProgramParams_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -2097,5 +2102,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreHardwareBuffer.h b/OgreMain/include/OgreHardwareBuffer.h
index 729bd3046..3ddc80bf6 100644
--- a/OgreMain/include/OgreHardwareBuffer.h
+++ b/OgreMain/include/OgreHardwareBuffer.h
@@ -27,6 +27,10 @@ THE SOFTWARE.
 */
 #ifndef __HardwareBuffer__
 #define __HardwareBuffer__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -432,6 +436,8 @@ namespace Ogre {
     /** @} */
     /** @} */
 }
+#ifdef _WIN32
+#pragma warning(pop)
 #endif
 
-
+#endif
diff --git a/OgreMain/include/OgreInstanceBatch.h b/OgreMain/include/OgreInstanceBatch.h
index ece52a464..b457e42f8 100644
--- a/OgreMain/include/OgreInstanceBatch.h
+++ b/OgreMain/include/OgreInstanceBatch.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __InstanceBatch_H__
 #define __InstanceBatch_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreRenderOperation.h"
@@ -364,5 +369,9 @@ namespace Ogre
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __InstanceBatch_H__
diff --git a/OgreMain/include/OgreInstanceManager.h b/OgreMain/include/OgreInstanceManager.h
index ea5f56d1e..4b8d8d0a0 100644
--- a/OgreMain/include/OgreInstanceManager.h
+++ b/OgreMain/include/OgreInstanceManager.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __InstanceManager_H__
 #define __InstanceManager_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreRenderOperation.h"
@@ -310,5 +315,9 @@ namespace Ogre
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __InstanceManager_H__
diff --git a/OgreMain/include/OgreInstancedEntity.h b/OgreMain/include/OgreInstancedEntity.h
index 27f9592d1..4dd1ea205 100644
--- a/OgreMain/include/OgreInstancedEntity.h
+++ b/OgreMain/include/OgreInstancedEntity.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __InstancedEntity_H__
 #define __InstancedEntity_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgreMovableObject.h"
 #include "OgreNode.h"
@@ -294,5 +299,9 @@ namespace Ogre
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreKeyFrame.h b/OgreMain/include/OgreKeyFrame.h
index f9e2a1054..04d9e373f 100644
--- a/OgreMain/include/OgreKeyFrame.h
+++ b/OgreMain/include/OgreKeyFrame.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __KeyFrame_H__
 #define __KeyFrame_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreVector.h"
@@ -256,5 +261,9 @@ namespace Ogre
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreLight.h b/OgreMain/include/OgreLight.h
index 371a605a6..e03d9090a 100644
--- a/OgreMain/include/OgreLight.h
+++ b/OgreMain/include/OgreLight.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef _LIGHT_H__
 #define _LIGHT_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -638,4 +643,8 @@ namespace Ogre {
 #include "OgreHeaderSuffix.h"
 
 } // namespace Ogre
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif // _LIGHT_H__
diff --git a/OgreMain/include/OgreManualObject.h b/OgreMain/include/OgreManualObject.h
index 5fa5fd448..5317fbdd0 100644
--- a/OgreMain/include/OgreManualObject.h
+++ b/OgreMain/include/OgreManualObject.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __OgreManualObject_H__
 #define __OgreManualObject_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMovableObject.h"
@@ -736,6 +741,8 @@ namespace Ogre
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
 #endif
 
-
+#endif
diff --git a/OgreMain/include/OgreMath.h b/OgreMain/include/OgreMath.h
index 200146266..a2cc5c894 100644
--- a/OgreMain/include/OgreMath.h
+++ b/OgreMain/include/OgreMath.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Math_H__
 #define __Math_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreHeaderPrefix.h"
@@ -819,5 +824,9 @@ namespace Ogre
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreMatrix3.h b/OgreMain/include/OgreMatrix3.h
index b204bc439..94bab86b2 100644
--- a/OgreMain/include/OgreMatrix3.h
+++ b/OgreMain/include/OgreMatrix3.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Matrix3_H__
 #define __Matrix3_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -333,4 +338,8 @@ namespace Ogre
     /** @} */
     /** @} */
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreMatrix4.h b/OgreMain/include/OgreMatrix4.h
index 346f35d8b..83a7ed903 100644
--- a/OgreMain/include/OgreMatrix4.h
+++ b/OgreMain/include/OgreMatrix4.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Matrix4__
 #define __Matrix4__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -598,4 +603,8 @@ namespace Ogre
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreMesh.h b/OgreMain/include/OgreMesh.h
index 23453a867..ac2ac7e87 100644
--- a/OgreMain/include/OgreMesh.h
+++ b/OgreMain/include/OgreMesh.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Mesh_H__
 #define __Mesh_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -1039,5 +1044,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __Mesh_H__
diff --git a/OgreMain/include/OgreMeshManager.h b/OgreMain/include/OgreMeshManager.h
index 99542ffb1..8b6617d3e 100644
--- a/OgreMain/include/OgreMeshManager.h
+++ b/OgreMain/include/OgreMeshManager.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __MeshManager_H__
 #define __MeshManager_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -482,5 +487,9 @@ namespace Ogre {
 } //namespace
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreMovablePlane.h b/OgreMain/include/OgreMovablePlane.h
index baa764edd..4b971c67a 100644
--- a/OgreMain/include/OgreMovablePlane.h
+++ b/OgreMain/include/OgreMovablePlane.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __MovablePlane_H__
 #define __MovablePlane_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -96,5 +101,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreNode.h b/OgreMain/include/OgreNode.h
index 72bcf8432..e32fdc0ab 100644
--- a/OgreMain/include/OgreNode.h
+++ b/OgreMain/include/OgreNode.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef _Node_H__
 #define _Node_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -682,5 +687,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // _Node_H__
diff --git a/OgreMain/include/OgreOptimisedUtil.h b/OgreMain/include/OgreOptimisedUtil.h
index c50407965..82a0f111c 100644
--- a/OgreMain/include/OgreOptimisedUtil.h
+++ b/OgreMain/include/OgreOptimisedUtil.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __OptimisedUtil_H__
 #define __OptimisedUtil_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreEdgeListBuilder.h"
@@ -241,5 +246,9 @@ namespace Ogre {
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif  // __OptimisedUtil_H__
diff --git a/OgreMain/include/OgreParticle.h b/OgreMain/include/OgreParticle.h
index 06d0e9b7f..8e7a7dd92 100644
--- a/OgreMain/include/OgreParticle.h
+++ b/OgreMain/include/OgreParticle.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Particle_H__
 #define __Particle_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreHeaderPrefix.h"
@@ -144,5 +149,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreParticleEmitter.h b/OgreMain/include/OgreParticleEmitter.h
index 67f13e43f..fb4a9f629 100644
--- a/OgreMain/include/OgreParticleEmitter.h
+++ b/OgreMain/include/OgreParticleEmitter.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __ParticleEmitter_H__
 #define __ParticleEmitter_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreVector.h"
@@ -524,5 +529,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreParticleSystem.h b/OgreMain/include/OgreParticleSystem.h
index 1144ebdb1..2e091dd05 100644
--- a/OgreMain/include/OgreParticleSystem.h
+++ b/OgreMain/include/OgreParticleSystem.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __ParticleSystem_H__
 #define __ParticleSystem_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -885,5 +890,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePass.h b/OgreMain/include/OgrePass.h
index 6caa43631..c38b54a3a 100644
--- a/OgreMain/include/OgrePass.h
+++ b/OgreMain/include/OgrePass.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Pass_H__
 #define __Pass_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreColourValue.h"
@@ -1669,5 +1674,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePatchSurface.h b/OgreMain/include/OgrePatchSurface.h
index 1b3e21db0..b6c910b44 100644
--- a/OgreMain/include/OgrePatchSurface.h
+++ b/OgreMain/include/OgrePatchSurface.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __PatchSurface_H__
 #define __PatchSurface_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -238,5 +243,9 @@ namespace Ogre {
 } // namespace
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePlane.h b/OgreMain/include/OgrePlane.h
index ebb3f703b..955c0c215 100644
--- a/OgreMain/include/OgrePlane.h
+++ b/OgreMain/include/OgrePlane.h
@@ -35,6 +35,11 @@ THE SOFTWARE.
 
 #ifndef __Plane_H__
 #define __Plane_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -277,5 +282,9 @@ namespace Ogre {
     /** @} */
 
 } // namespace Ogre
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePlaneBoundedVolume.h b/OgreMain/include/OgrePlaneBoundedVolume.h
index 3d4c08e34..408bf70ef 100644
--- a/OgreMain/include/OgrePlaneBoundedVolume.h
+++ b/OgreMain/include/OgrePlaneBoundedVolume.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __PlaneBoundedVolume_H_
 #define __PlaneBoundedVolume_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -134,5 +139,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgrePolygon.h b/OgreMain/include/OgrePolygon.h
index 7528f4141..1997ab6a5 100644
--- a/OgreMain/include/OgrePolygon.h
+++ b/OgreMain/include/OgrePolygon.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 */
 #ifndef __Polygon_H__
 #define __Polygon_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreVector.h"
@@ -146,5 +151,9 @@ namespace Ogre
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePose.h b/OgreMain/include/OgrePose.h
index 3a20408ce..30ee20c57 100644
--- a/OgreMain/include/OgrePose.h
+++ b/OgreMain/include/OgrePose.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __OGRE_POSE_H
 #define __OGRE_POSE_H
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -151,5 +156,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgrePrerequisites.h b/OgreMain/include/OgrePrerequisites.h
index 71a0395d5..e5381990a 100644
--- a/OgreMain/include/OgrePrerequisites.h
+++ b/OgreMain/include/OgrePrerequisites.h
@@ -25,6 +25,11 @@ THE SOFTWARE
 -------------------------------------------------------------------------*/
 #ifndef __OgrePrerequisites_H__
 #define __OgrePrerequisites_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Platform-specific stuff
 #include "OgrePlatform.h"
@@ -382,6 +387,9 @@ namespace Ogre
     };
 } // Ogre
 
-#endif // __OgrePrerequisites_H__
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
 
+#endif // __OgrePrerequisites_H__
 
diff --git a/OgreMain/include/OgreQuaternion.h b/OgreMain/include/OgreQuaternion.h
index c82bbdaf6..edfbaee45 100644
--- a/OgreMain/include/OgreQuaternion.h
+++ b/OgreMain/include/OgreQuaternion.h
@@ -35,6 +35,11 @@ THE SOFTWARE.
 
 #ifndef __Quaternion_H__
 #define __Quaternion_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMath.h"
@@ -367,5 +372,9 @@ namespace Ogre {
 
 
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif 
diff --git a/OgreMain/include/OgreRay.h b/OgreMain/include/OgreRay.h
index 1b44cee66..e689cabce 100644
--- a/OgreMain/include/OgreRay.h
+++ b/OgreMain/include/OgreRay.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Ray_H_
 #define __Ray_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -152,4 +157,8 @@ namespace Ogre {
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreRectangle2D.h b/OgreMain/include/OgreRectangle2D.h
index b3f58eff2..9037bd811 100644
--- a/OgreMain/include/OgreRectangle2D.h
+++ b/OgreMain/include/OgreRectangle2D.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef _Rectangle2D_H__
 #define _Rectangle2D_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -96,6 +101,8 @@ namespace Ogre {
 
 }// namespace
 
+#ifdef _WIN32
+#pragma warning(pop)
 #endif
 
-
+#endif
diff --git a/OgreMain/include/OgreRenderSystem.h b/OgreMain/include/OgreRenderSystem.h
index f81349ca0..1ef1be595 100644
--- a/OgreMain/include/OgreRenderSystem.h
+++ b/OgreMain/include/OgreRenderSystem.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __RenderSystem_H_
 #define __RenderSystem_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -1316,5 +1321,9 @@ namespace Ogre
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreRenderable.h b/OgreMain/include/OgreRenderable.h
index 82f9be8e9..85edc5a50 100644
--- a/OgreMain/include/OgreRenderable.h
+++ b/OgreMain/include/OgreRenderable.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Renderable_H__
 #define __Renderable_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -385,4 +390,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif //__Renderable_H__
diff --git a/OgreMain/include/OgreSceneManager.h b/OgreMain/include/OgreSceneManager.h
index 3dfcf98ea..49775daab 100644
--- a/OgreMain/include/OgreSceneManager.h
+++ b/OgreMain/include/OgreSceneManager.h
@@ -29,6 +29,11 @@ Torus Knot Software Ltd.
 -------------------------------------------------------------------------*/
 #ifndef __SceneManager_H__
 #define __SceneManager_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -3624,5 +3629,9 @@ namespace Ogre {
 } // Namespace
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreSceneNode.h b/OgreMain/include/OgreSceneNode.h
index 87e1e9d1e..e9140ed36 100644
--- a/OgreMain/include/OgreSceneNode.h
+++ b/OgreMain/include/OgreSceneNode.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef _SceneNode_H__
 #define _SceneNode_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -457,5 +462,9 @@ namespace Ogre {
 }// namespace
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreSceneQuery.h b/OgreMain/include/OgreSceneQuery.h
index 2eecd1bc8..8c57438ff 100644
--- a/OgreMain/include/OgreSceneQuery.h
+++ b/OgreMain/include/OgreSceneQuery.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __SceneQuery_H__
 #define __SceneQuery_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreSphere.h"
@@ -555,5 +560,9 @@ namespace Ogre {
 }
     
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreSerializer.h b/OgreMain/include/OgreSerializer.h
index 311b2808a..2ba20b23d 100644
--- a/OgreMain/include/OgreSerializer.h
+++ b/OgreMain/include/OgreSerializer.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __Serializer_H__
 #define __Serializer_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreHeaderPrefix.h"
@@ -128,5 +133,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreShadowCameraSetupFocused.h b/OgreMain/include/OgreShadowCameraSetupFocused.h
index 2a41f0d53..6947f2f60 100644
--- a/OgreMain/include/OgreShadowCameraSetupFocused.h
+++ b/OgreMain/include/OgreShadowCameraSetupFocused.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 */
 #ifndef __ShadowCameraSetupFocused_H__
 #define __ShadowCameraSetupFocused_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreShadowCameraSetup.h"
@@ -312,5 +317,9 @@ namespace Ogre {
 } // namespace Ogre
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __ShadowCameraSetupFocused_H__
diff --git a/OgreMain/include/OgreShadowCameraSetupLiSPSM.h b/OgreMain/include/OgreShadowCameraSetupLiSPSM.h
index 3c65077e4..5d04739e7 100644
--- a/OgreMain/include/OgreShadowCameraSetupLiSPSM.h
+++ b/OgreMain/include/OgreShadowCameraSetupLiSPSM.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 */
 #ifndef __ShadowCameraSetupLiSPSM_H__
 #define __ShadowCameraSetupLiSPSM_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreShadowCameraSetupFocused.h"
@@ -246,5 +251,9 @@ namespace Ogre
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreShadowCameraSetupPlaneOptimal.h b/OgreMain/include/OgreShadowCameraSetupPlaneOptimal.h
index 5abd5ac81..e23916f2a 100644
--- a/OgreMain/include/OgreShadowCameraSetupPlaneOptimal.h
+++ b/OgreMain/include/OgreShadowCameraSetupPlaneOptimal.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __ShadowCameraSetupPlaneOptimal_H__
 #define __ShadowCameraSetupPlaneOptimal_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreShadowCameraSetup.h"
@@ -91,5 +96,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreShadowCaster.h b/OgreMain/include/OgreShadowCaster.h
index 2d5b3b978..73117c412 100644
--- a/OgreMain/include/OgreShadowCaster.h
+++ b/OgreMain/include/OgreShadowCaster.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __ShadowCaster_H__
 #define __ShadowCaster_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreRenderable.h"
@@ -268,5 +273,9 @@ namespace Ogre {
     /** @} */
 } // namespace Ogre
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif // __ShadowCaster_H__
diff --git a/OgreMain/include/OgreSimpleSpline.h b/OgreMain/include/OgreSimpleSpline.h
index 53307d054..484545649 100644
--- a/OgreMain/include/OgreSimpleSpline.h
+++ b/OgreMain/include/OgreSimpleSpline.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __SimpleSpline_H__
 #define __SimpleSpline_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMatrix4.h"
@@ -135,5 +140,9 @@ namespace Ogre {
 #include "OgreHeaderSuffix.h"
 
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreSkeletonFileFormat.h b/OgreMain/include/OgreSkeletonFileFormat.h
index bf5bcd70e..e33bc460d 100644
--- a/OgreMain/include/OgreSkeletonFileFormat.h
+++ b/OgreMain/include/OgreSkeletonFileFormat.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __SkeletonFileFormat_H__
 #define __SkeletonFileFormat_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -113,5 +118,9 @@ namespace Ogre {
 
 } // namespace
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreSkeletonInstance.h b/OgreMain/include/OgreSkeletonInstance.h
index 8e2482c11..66ca9c768 100644
--- a/OgreMain/include/OgreSkeletonInstance.h
+++ b/OgreMain/include/OgreSkeletonInstance.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __SkeletonInstance_H__
 #define __SkeletonInstance_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreSkeleton.h"
@@ -158,5 +163,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreSphere.h b/OgreMain/include/OgreSphere.h
index e1b7b8afb..7b6e1517d 100644
--- a/OgreMain/include/OgreSphere.h
+++ b/OgreMain/include/OgreSphere.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Sphere_H_
 #define __Sphere_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 // Precompiler options
 #include "OgrePrerequisites.h"
@@ -134,5 +139,9 @@ namespace Ogre {
 
 }
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreStaticGeometry.h b/OgreMain/include/OgreStaticGeometry.h
index eff3e95a0..307ed39b3 100644
--- a/OgreMain/include/OgreStaticGeometry.h
+++ b/OgreMain/include/OgreStaticGeometry.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __StaticGeometry_H__
 #define __StaticGeometry_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreMovableObject.h"
@@ -784,5 +789,9 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreStreamSerialiser.h b/OgreMain/include/OgreStreamSerialiser.h
index a4d13e535..37a854719 100644
--- a/OgreMain/include/OgreStreamSerialiser.h
+++ b/OgreMain/include/OgreStreamSerialiser.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __StreamSerialiser_H__
 #define __StreamSerialiser_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -392,5 +397,9 @@ namespace Ogre
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreStringConverter.h b/OgreMain/include/OgreStringConverter.h
index a3e4636a3..ecbd11af2 100644
--- a/OgreMain/include/OgreStringConverter.h
+++ b/OgreMain/include/OgreStringConverter.h
@@ -28,6 +28,11 @@ THE SOFTWARE.
 
 #ifndef __StringConverter_H__
 #define __StringConverter_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgreCommon.h"
 #include "OgrePrerequisites.h"
@@ -356,5 +361,9 @@ namespace Ogre {
 
 
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
 
diff --git a/OgreMain/include/OgreSubMesh.h b/OgreMain/include/OgreSubMesh.h
index 0e6426ce3..c597be05a 100644
--- a/OgreMain/include/OgreSubMesh.h
+++ b/OgreMain/include/OgreSubMesh.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __SubMesh_H_
 #define __SubMesh_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 
@@ -276,6 +281,8 @@ namespace Ogre {
 
 #include "OgreHeaderSuffix.h"
 
+#ifdef _WIN32
+#pragma warning(pop)
 #endif
 
-
+#endif
diff --git a/OgreMain/include/OgreTangentSpaceCalc.h b/OgreMain/include/OgreTangentSpaceCalc.h
index 783d91304..362fb5151 100644
--- a/OgreMain/include/OgreTangentSpaceCalc.h
+++ b/OgreMain/include/OgreTangentSpaceCalc.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef _OgreTangentSpaceCalc_H_
 #define _OgreTangentSpaceCalc_H_
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreRenderOperation.h"
@@ -263,5 +268,9 @@ namespace Ogre
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/include/OgreTextureManager.h b/OgreMain/include/OgreTextureManager.h
index 6235951c2..822c7bd73 100644
--- a/OgreMain/include/OgreTextureManager.h
+++ b/OgreMain/include/OgreTextureManager.h
@@ -28,6 +28,10 @@ THE SOFTWARE.
 #ifndef _TextureManager_H__
 #define _TextureManager_H__
 
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
 
 #include "OgrePrerequisites.h"
 
@@ -456,4 +460,8 @@ namespace Ogre {
     /** @} */
 }// Namespace
 
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreVector.h b/OgreMain/include/OgreVector.h
index 2e08a5414..c842987af 100644
--- a/OgreMain/include/OgreVector.h
+++ b/OgreMain/include/OgreVector.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Vector_H__
 #define __Vector_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 
 #include "OgrePrerequisites.h"
@@ -910,4 +915,8 @@ namespace Ogre
     /** @} */
 
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 #endif
diff --git a/OgreMain/include/OgreViewport.h b/OgreMain/include/OgreViewport.h
index 72d68d97f..c41a561f0 100644
--- a/OgreMain/include/OgreViewport.h
+++ b/OgreMain/include/OgreViewport.h
@@ -27,6 +27,11 @@ THE SOFTWARE.
 */
 #ifndef __Viewport_H__
 #define __Viewport_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgrePrerequisites.h"
 #include "OgreCommon.h"
@@ -444,5 +449,9 @@ namespace Ogre {
 }
 
 #include "OgreHeaderSuffix.h"
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #endif
diff --git a/OgreMain/src/OgreStableHeaders.h b/OgreMain/src/OgreStableHeaders.h
index 071039a86..efe8859fc 100644
--- a/OgreMain/src/OgreStableHeaders.h
+++ b/OgreMain/src/OgreStableHeaders.h
@@ -34,6 +34,11 @@ THE SOFTWARE.
 
 #ifndef __OgreStableHeaders__
 #define __OgreStableHeaders__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
 
 #include "OgreConfig.h"
 #include "OgreExports.h"
@@ -107,6 +112,10 @@ THE SOFTWARE.
 #if OGRE_NO_ZIP_ARCHIVE == 0
 #   include "OgreZip.h"
 #endif
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
 
 #if OGRE_COMPILER == OGRE_COMPILER_MSVC
 #define OGRE_IGNORE_DEPRECATED_BEGIN __pragma(warning(push)) \
diff --git a/PlugIns/OctreeZone/CMakeLists.txt b/PlugIns/OctreeZone/CMakeLists.txt
--- a/PlugIns/OctreeZone/CMakeLists.txt	2019-06-24 23:04:20.000000000 +0000
+++ b/PlugIns/OctreeZone/CMakeLists.txt	2019-10-04 16:58:03.669588969 +0000
@@ -23,6 +23,11 @@ generate_export_header(Plugin_OctreeZone
     EXPORT_MACRO_NAME _OgreOctreeZonePluginExport
     EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/OgreOctreeZonePrerequisites.h)
 
+if (UNIX)
+  set_property(TARGET Plugin_OctreeZone APPEND PROPERTY
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/${OGRE_LIB_DIRECTORY}/OGRE)
+endif ()
+
 ogre_config_framework(Plugin_OctreeZone)
 
 ogre_config_plugin(Plugin_OctreeZone)
diff --git a/OgreMain/include/OgreTimer.h b/OgreMain/include/OgreTimer.h
--- a/OgreMain/include/OgreTimer.h	2019-06-24 16:04:20.000000000 -0700
+++ b/OgreMain/include/OgreTimer.h	2019-10-10 12:33:07.243621900 -0700
@@ -28,6 +28,10 @@ THE SOFTWARE.
 
 #ifndef __OGRE_TIMER_H__
 #define __OGRE_TIMER_H__
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
 
 #include "OgrePrerequisites.h"
 #include <chrono>
@@ -59,4 +63,7 @@ namespace Ogre
         uint64_t getMicrosecondsCPU();
     };
 }
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
 #endif

