diff --git a/OgreMain/include/OgreRenderSystem.h b/OgreMain/include/OgreRenderSystem.h
index cf82fdc0f..711228e3a 100644
--- a/OgreMain/include/OgreRenderSystem.h
+++ b/OgreMain/include/OgreRenderSystem.h
@@ -971,12 +971,12 @@ namespace Ogre
         <i>pixels</i>.
         */
         virtual void setScissorTest(bool enabled, const Rect& rect = Rect()) = 0;
-        /// @deprecated
-        OGRE_DEPRECATED void setScissorTest(bool enabled, size_t left, size_t top = 0,
-                                            size_t right = 800, size_t bottom = 600)
-        {
-            setScissorTest(enabled, Rect(left, top, right, bottom));
-        }
+        // /// @deprecated
+        // OGRE_DEPRECATED void setScissorTest(bool enabled, size_t left, size_t top = 0,
+        //                                     size_t right = 800, size_t bottom = 600)
+        // {
+        //     setScissorTest(enabled, Rect(left, top, right, bottom));
+        // }
 
         /** Clears one or more frame buffers on the active render target. 
         @param buffers Combination of one or more elements of FrameBufferType
